core/src/token_info/cached.rs

```diff
@@ -141,6 +154,24 @@ impl TokenInfoFetching for TokenInfoCache {
     fn all_ids<'a>(&'a self) -> BoxFuture<'a, Result<Vec<TokenId>>> {
         self.inner.all_ids()
     }
+
+    fn find_token_by_symbol<'a>(
+        &'a self,
+        symbol: &'a str,
+    ) -> BoxFuture<'a, Result<Option<(TokenId, TokenBaseInfo)>>> {
+        async move {
+            // NOTE: First check the cache directly before refetching. This
+            // allows us to exit early without checking to see if there are
+            // missing token infos that we need to cache.
+            if let Some(result) = self.find_cached_token_by_symbol(symbol).await {
+                return Ok(Some(result));
+            }
+
+            self.cache_all(1).await?;
```

Comment:
Why 1 here (I'm not a huge fan of this unnamed argument at least when reviewing)?

Reply:
I'll name it! Its the number of concurrent requests.